sympy/core/tests/test_arit.py

```diff
@@ -1996,6 +1996,8 @@ def test_Mod():
     from sympy.abc import phi
     assert Mod(4.0*Mod(phi, 1) , 2) == 2.0*(Mod(2*(Mod(phi, 1)), 1))
 
+    assert Mod(3*x, 2) == Mod(3*x, 2)
```

Comment:
This test does not do anything: it would already pass without the changes here.

Reply:
The test should be
```
assert unchanged(Mod, 3*x, 2)
```

Reply:
There should also be a test for what happens with a symbol that has `integer=True` set (if there is not one already).

Reply:
```python
x = symbols('x', integer=True)
unchanged(Mod, 3*x, 2) 
``` 
Give output False, as [mentioned ](https://github.com/sympy/sympy/issues/26016#issuecomment-1873350941)here 
> This simplification assumes that x is an integer which is not necessarily the case if no assumptions are set on x:

Reply:
The test should be:
```
xi = symbols('x', integer=True)
assert unchanged(Mod, xi, 2) 
assert Mod(3*xi, 2) == Mod(xi, 2)
```